[ANALYZER-3942] Add Intl API support to CGG's JavaScript environment and fix localized string resources

dcleao opened the pull request:
This PR makes the following changes:
- `WebResourceLoader` to pass Pentaho session's locale override via trusted proxy arguments; this allows locale sensitive resources, such as `/i18n`, to be more faithfully proxied.
- Upgraded Rhino to version 1.7.13 which already supports many ES6 features.
- Removed JS shims which are no longer needed in the new Rhino version
- Added the `js-intl-polyfill` module which creates the `Intl` polyfill used by CGG
- Removed `useGlobal` flag and corresponding functionality from the CGG JS environment (was never used); CGG is always installed on the global scope.

This PR depends on the following PRs:
- https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1529
- https://github.com/webdetails/ccc/pull/378
- https://github.com/pentaho/pentaho-platform/pull/4873

I added some background information that you might find relevant at: https://jira.pentaho.com/browse/ANALYZER-3942?focusedCommentId=407628&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-407628.

@davidmsantos90, @afrjorge, @bennychow please review.
/cc @jcarneirohv

buildguy commented:
### :x: Build failed in 1m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

davidmsantos90 commented on core/src/main/javascript/pt/webdetails/cgg/resources/cgg/interop.js:
```diff
@@ -10,11 +10,9 @@
+define(function() {
 
-    var javaLang = util.global.java.lang;
+    var javaLang = java.lang;
```
is `java` a global variable or did you miss `globalThis` on this one?

dcleao replied:
Yes, `java` is a global variable. And I've relaxed using `globalThis` everywhere.

davidmsantos90 commented on js-intl-polyfill/pom.xml:
```diff
@@ -0,0 +1,51 @@
+  <packaging>pom</packaging>
+
+  <properties>
+    <nodejs.version>v15.1.0</nodejs.version>
```
do you need this specific version? it would be better to use a LTS version

dcleao replied:
Good point.
NodeJS is used at build time only (there are no tests).
The version of NodeJS inherited via the parent POM is `v10.0.0`.
While Rollup itself would work with NodeJS `v10.12.0`, our `rollup.config.js` script uses functionality only available since `v12.2.0`.

Now why `v15.1.0`? It seems, ... that's the version I have installed on my machine :-/
And I ended up not revisiting this choice.

The latest LTS version is `v14.17.0`, so I'll change to that one.

Thanks!

buildguy commented:
### :x: Build failed in 1m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

buildguy commented:
### :x: Build failed in 1m 2s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

buildguy commented:
### :white_check_mark: Build finished in 2m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **2**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/cgg-plugin/job/PR-148/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

buildguy commented:
### :white_check_mark: Build finished in 2m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **2**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/cgg-plugin/job/PR-148/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

buildguy commented:
### :white_check_mark: Build finished in 3m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **2**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/cgg-plugin/job/PR-148/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

dcleao commented:
@davidmsantos90, @afrjorge, @bennychow

Given all the dependencies are now merged, this PR already builds.
I believe this is now ready for your final OK.

I have fixed an issue with the build.
I have addressed your review comments.
I have addressed Sonar's complaints, except one Code Smell. Please, see if the justification I gave (in Sonar) is satisfactory.

Thanks!

davidmsantos90 reviewed (approved):
👍🏼

dcleao commented:
Thanks guys!
